### PR TITLE
spring boot debugging cleanup and fix wfs missing param

### PR DESCRIPTION
**Pull request by rhoads-zach**
@metacosm let me know what you think.

### CODE REVIEW

File: docs/topics/proc_starting-an-uberjar-in-debugging-mode.adoc [hunk 1]
```diff
@@ -19,15 +19,14 @@ If you chose to package your application as a {runtime} uberjar, debug it by exe
...
 ----
-$ java -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=$PORT_NUMBER -jar $UBERJAR_FILENAME
+$ java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=$PORT_NUMBER -jar $UBERJAR_FILENAME
 ----
-
-Here, `$PORT_NUMBER` is an unused port number of your choice.
-Remember this number for the remote debugger configuration.
---
++
+`$PORT_NUMBER` is an unused port number of your choice. Remember this number for the remote debugger configuration.
```
Comment (Ladicek) [inline docs/topics/proc_starting-an-uberjar-in-debugging-mode.adoc]: Nit: multiple sentences on a single line :-)

File: docs/topics/proc_starting-an-uberjar-in-debugging-mode.adoc [hunk 2]
```diff
@@ -19,15 +19,14 @@ If you chose to package your application as a {runtime} uberjar, debug it by exe
...
++
+If you want the application to pause immediately, change `suspend` to `y`.
```
Comment (Ladicek) [inline docs/topics/proc_starting-an-uberjar-in-debugging-mode.adoc]: "Immediately" is very vague here. I'd rephrase to something like "If you want the JVM to pause and wait for remote debugger connection before it starts the application".

File: docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc [hunk 3]
```diff
@@ -1,13 +1,27 @@
+// This is a parameterized module. Parameters used:
+//
+//   springboot-debug: Signifies whether this is used by spring boot or not. 
+//
+// Rationale: This procedure is identical in Swarm and Vert.x, and varies slightly in spring boot.
```
Comment (Ladicek) [inline docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc]: This is just not true and whoever told you so was wrong. All of Swarm, Vert.x and Spring Boot use the _same_ OpenJDK image, and so can use the exact same instructions.

Snippet from `/opt/run-java/debug-options`:

```bash
  if [ "x${JAVA_ENABLE_DEBUG}" != "x" -o "x${JAVA_DEBUG_ENABLE}" != "x" -o "x${JAVA_DEBUG}" != "x" ]; then
    local debug_port=${JAVA_DEBUG_PORT:-5005}
    echo "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=${debug_port}"
  fi
```

So it's entirely possible to use `JAVA_DEBUG`, `JAVA_ENABLE_DEBUG` and `JAVA_DEBUG_ENABLE` interchangeably. I just picked `JAVA_DEBUG` because it's shortest, but they are equivalent.

File: docs/topics/proc_starting-your-spring-boot-application-locally-in-debugging-mode.adoc [hunk 4]
```diff
@@ -13,13 +13,14 @@ This method is applicable at least when launching the application manually using
...
---
 [source,bash,options="nowrap"]
 ----
-$ mvn spring-boot:run -Drun.jvmArguments="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=$PORT_NUMBER"
+$ mvn spring-boot:run -Drun.jvmArguments="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=$PORT_NUMBER"
 ----
++
+`$PORT_NUMBER` is an unused port number of your choice. Remember this number for the remote debugger configuration.
++
+If you want the application to pause immediately, change `suspend` to `y`.
```
Comment (Ladicek) [inline docs/topics/proc_starting-your-spring-boot-application-locally-in-debugging-mode.adoc]: Same comments as for `proc_starting-an-uberjar-in-debugging-mode.adoc`.

File: docs/wf-swarm-runtime/master.adoc [hunk 5]
```diff
@@ -283,7 +283,11 @@ This sections contains information about debugging your {WildFlySwarm}&#x2013;ba
 
 To remotely debug an application, you must first configure it to start in a debugging mode, and then attach a debugger to it.
 
+:parameter-debug-property: swarm.debug.port
+:parameter-maven-goal: wildfly-swarm:run
 include::topics/proc_starting-your-application-locally-in-debugging-mode.adoc[leveloffset=+3]
+:parameter-debug-property!: 
+:parameter-maven-goal:
```
Comment (Ladicek) [inline docs/wf-swarm-runtime/master.adoc]: I don't really understand this, but why does one line contain `!` and the other one doesn't? :-)

File: docs/topics/proc_starting-an-uberjar-in-debugging-mode.adoc (hunk 2 above)
Comment (metacosm) [reply docs/topics/proc_starting-an-uberjar-in-debugging-mode.adoc]: +1

File: docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc (hunk 3 above)
Comment (metacosm) [reply docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc]: As an added comment, maybe we should also document that the default debugging port can be changed by passing a different value to `JAVA_DEBUG_PORT`?

File: docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc [hunk 6]
```diff
@@ -18,11 +32,11 @@ To debug your {runtime}-based application on OpenShift remotely, you must set th
 $ oc get dc
 ----
 
-. Set the `JAVA_DEBUG` environment variable in the deployment configuration of your application to `true`, which configures the JVM to open the port number `5005` for debugging. For example:
+. Set the `{debug-var}` environment variable in the deployment configuration of your application to `true`, which configures the JVM to open the port number `5005` for debugging. For example:
```
Comment (metacosm) [inline docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc]: It'd make sense to document changing the port number here as well via `JAVA_DEBUG_PORT`…

File: docs/wf-swarm-runtime/master.adoc (hunk 5 above)
Comment (metacosm) [reply docs/wf-swarm-runtime/master.adoc]: Same here.

File: docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc (hunk 3 above)
Comment (Ladicek) [reply docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc]: I don't think you often need to change the debugger port when running the JVM in a container, but at the same time, I agree this can be useful to know. So I don't really mind, but I certainly wouldn't disagree :-)

File: docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc (hunk 6 above)
Comment (Ladicek) [reply docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc]: I'd just document `JAVA_DEBUG_PORT` on a single place, not two.

File: docs/wf-swarm-runtime/master.adoc (hunk 5 above)
Comment (rhoads-zach) [reply docs/wf-swarm-runtime/master.adoc]: Nice catch, i missed that one.

**Comment by rhoads-zach** [comment]
thank you for all the feedback @Ladicek and @metacosm 

Updated per your feedback.

Let me know what you think.
